integreat_cms/cms/utils/linkcheck_utils.py

```diff
@@ -37,12 +37,17 @@ def get_urls(region_slug=None, url_ids=None, prefetch_content_objects=True):
         urls = urls.filter(id__in=url_ids)
     if region_slug:
         # Get all link objects of the requested region
-        region_links = Link.objects.filter(
+        all_region_links = Link.objects.filter(
             Q(page_translation__page__region__slug=region_slug)
             | Q(imprint_translation__page__region__slug=region_slug)
             | Q(event_translation__event__region__slug=region_slug)
             | Q(poi_translation__poi__region__slug=region_slug)
         ).order_by("id")
+        region_link_ids = []
+        for region_link in all_region_links:
+            if region_link.content_object == region_link.content_object.latest_version:
+                region_link_ids += [region_link.id]
```

Comment:
Hmm, I'm not too convinced about this loop, this function is too slow already (see #2539) and this loop will cause a lot of additional queries.
Maybe we could calculate the ids of the latest versions in one query beforehand, similar to how we did in the dashboard view? Or find another way how we could move this condition into the sql query itself?

Since we need to improve the performance of this function anyway, we could also dive deeper into what's costing the most time here and maybe refactor the entire function.

Reply:
I'm not confident but changed the behaivor so IDs of latest versions of each content type will be collected and then used in the query for `region_link`